### ISSUE TITLE
How to record multiple audio streams?

### ISSUE THREAD

**Issue by xiaos** [bytedeco/javacv#466]
I am using FFmpegFrameRecorder to record 1 video and multiple audios, any suggestion to develop this?

**Comment by saudet** [comment]
Sounds like a great enhancement. Want to give it a try?

**Comment by xiaos** [comment]
yes, want to try, please give some tips, thanks :)

**Comment by saudet** [comment]
Look inside FFmpegFrameRecorder.java. It currently assumes only one audio/video stream as per the video_st and audio_st fields. Try to use more!

**Comment by xiaos** [comment]
Thanks, I will check.

**Comment by xiaos** [comment]
@saudet  hi, for this enhancement, do you need change the c/c++ file and compile? or java modify the java code? can give more tips? thanks.

**Comment by saudet** [comment]
No need to modify anything in C/C++. We just need to modify FFmpegFrameRecorder.java.

**Comment by xiaos** [comment]
thank you so much.

**Comment by xiaos** [comment]
@saudet,

I am working on this enhancement, just want to ask more information.

Inside the FFmpegFrameRecorder.java, I will add another audio_st, do i need add another copy of the audio related variables?

e.g
samples_in
samples_output

samples_in_ptr
samples_out_ptr

audio_outbuf

audio_outbuf_size
audio_input_frame_size

audio_codec.

Or I can share some of these variables? any tips? thank you.

**Comment by saudet** [comment]
yes, most likely, and we would want to support more than 2 streams, so we
could wrap those in an inner class to make things easier

**Comment by xiaos** [comment]
@saudet, 

multiple audio stream, but record one by one in order, so I think some variables can be shared, like 

audio_outbuf
samples_out
samples_in
samples_in_ptr
samples_out_ptr

**Comment by saudet** [comment]
try it!

**Comment by xiaos** [comment]
I have tried, two set of audio stream, two set of audio codec, then i record one by one,

it seems the later one always overwrite the first audio stream, any idea?

**Comment by xiaos** [comment]
thank you.

**Comment by saudet** [comment]
I'm sorry, it looks like I forgot to reply. Could you elaborate on what you mean by "overwrite"? BTW, if you could create a fork of the repository on GitHub, it would help me follow your progress. Thanks!

**Comment by tahaarian** [comment]
> @saudet,
> 
> multiple audio stream, but record one by one in order, so I think some variables can be shared, like
> 
> audio_outbuf
> samples_out
> samples_in
> samples_in_ptr
> samples_out_ptr

Hello @xiaos 

We need to record multiple audio streams same as you, may i ask you to share your final points ? 

We cant support two audios 

it could be great if you share your sample code with us 

Thank you in advance

Taha

**Comment by saudet** [comment]
@tahaarian If this is important to you, please consider paying for commercial support: https://xscode.com/bytedeco/javacv

**Comment by b005t3r** [comment]
My guess is recording multiple audio tracks is still not a thing? Or am I missing something?
